src/receivers/SocketModeFunctions.ts

```diff
@@ -0,0 +1,38 @@
+/* eslint-disable import/prefer-default-export */
+import type { Logger } from '@slack/logger';
+import { CodedError, ErrorCode } from '../errors';
+import { ReceiverEvent } from '../types';
+
+export class SocketModeFunctions {
+  // ------------------------------------------
+  // Error handlers for event processing
+  // ------------------------------------------
+
+  // The default processEventErrorHandler implementation:
+  // Developers can customize this behavior by passing processEventErrorHandler to the constructor
+  public static async defaultProcessEventErrorHandler(
+    args: SocketModeReceiverProcessEventErrorHandlerArgs,
+  ): Promise<boolean> {
+    const { error, logger, event } = args;
+    // TODO: more details like envelop_id, payload type etc. here
+    // To make them available, we need to enhance underlying SocketModeClient
+    // to return more properties to 'slack_event' listeners
+    logger.error(`An unhandled error occurred while Bolt processed a ${event.body.type} event: ${error}`);
```

Comment:
For Events API, the log message is grammatically incorrect.

```
[ERROR]   An unhandled error occurred while Bolt processed a event_callback event: Error: xxxx
```

Will change it to

```
[ERROR]   An unhandled error occurred while Bolt processed: (type: event_callback, error: Error: xxxx)
```